client/extensions/woocommerce/app/products/product-variation-types-form.js

```diff
@@ -0,0 +1,198 @@
+			} );
+			existingVariations.push( attributes.join( ' - ' ) );
+		} );
+
+		// Generate a list of new variations based on those that do not exist.
+		const newVariations = possibleVariations.filter( x => existingVariations.indexOf( x ) === -1 );
+
+		newVariations.map( function( newVariationName ) {
+			const attributes = Array();
+			newVariationName.split( ' - ' ).map( function( option, index ) {
+				const attribute = Array( 3 );
+				attribute.id = 0; // default for product level attributes in WC
+				attribute.name = indexesToLabels[ index ]; // Name (like 'Size' or 'Color')
+				attribute.option = option;
+				attributes.push( attribute );
+			} );
+			variations.push( {
+				name: newVariationName,
+				attributes,
+			} );
+		} );
+
+		return variations;
+	}
+
+	updateVariations() {
+		const { props } = this;
+		const newVariations = this.generateVariations( props.product.variationTypes, props.product.variations || [] );
+		newVariations.map( function( newVariation ) {
+			props.addVariation( null, newVariation );
+		} );
+	}
+
+	renderInputs( variation, index ) {
+		const _updateType = ( e ) => this.updateType( index, e );
+		const _updateValues = ( value ) => this.updateValues( index, value );
+		return (
+			<div key={ index } className="product-variation-types-form__fieldset">
+				<FormTextInput
+					placeholder={ i18n.translate( 'Color' ) }
+					value={ variation.type }
+					name="type"
+					onChange={ _updateType }
+					className="product-variation-types-form__field"
+				/>
+				<TokenField
+					placeholder={ i18n.translate( 'Comma separate these' ) }
+					value={ variation.values }
+					name="values"
+					onChange={ _updateValues }
+				/>
+			</div>
+		);
+	}
+
+	render() {
+		const { product } = this.props;
+		if ( ! product.variationTypes ) {
+			product.variationTypes = [ this.getNewFields() ];
+		}
+
+		let updateVariationsText = i18n.translate( 'Update Variations' );
+		if ( ! product.variations ) {
+			updateVariationsText = i18n.translate( 'Create Variations' );
+		}
+
+		const inputs = product.variationTypes.map( this.renderInputs, this );
+
+		return (
+			<div className="product-variation-types-form">
+				<strong>{ i18n.translate( 'Variation types' ) }</strong>
+				<p>
+					{ i18n.translate(
+						'Let\'s add some variations! A common {{em}}variation type{{/em}} is color. ' +
+						'The {{em}}values{{/em}} would be the colors the product is available in.',
+						{ components: { em: <em /> } }
+					) }
+				</p>
+
+				<div className="product-variation-types-form__group">
+					<div className="product-variation-types-form__labels">
+						<FormLabel className="product-variation-types-form__label">{ i18n.translate( 'Variation type' ) }</FormLabel>
+						<FormLabel>{ i18n.translate( 'Variation values' ) }</FormLabel>
+					</div>
+					{inputs}
+				</div>
+
+				<Button onClick={ this.addVariationType }>{ i18n.translate( 'Add another variation type' ) }</Button>
```

Comment:
Hi! I've found a possible matching string that has already been translated **1** time:
`translate( 'Add another variation' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).